### PR TITLE
Move logic for getting Xcode version to xcode-select-service

### PR DESCRIPTION
**Pull request by Mitko-Kerezov**
The logic is now used more than once and has to be extracted to a common place. Add test.
Ping @rosen-vladimirov

### CODE REVIEW

**Comment by justcodebuilduser** [comment]
:broken_heart:

**Comment by justcodebuilduser** [comment]
:heart:

**Comment by justcodebuilduser** [comment]
:heart:

File: declarations.d.ts [hunk 1]
```diff
@@ -1013,3 +1018,9 @@ interface ILiveSyncProvider {
 	 */
 	canExecuteFastSync(filePath: string, platform?: string): boolean;
 }
+
+interface IVersionData {
+	major: string;
+	minor: string;
+	patch: string;
+}
```
Comment (rosen-vladimirov) [inline declarations.d.ts]: new line please

**Comment by Mitko-Kerezov** [comment]
Closing and will re-target to `release`

**Pull request closed without merging**